### PR TITLE
Bug 1417857 Added more into the azure docs

### PR DESCRIPTION
**Pull request by bfallonf**
As per: https://bugzilla.redhat.com/show_bug.cgi?id=1417857

More work to come.

### CODE REVIEW

**Comment by tanaka-takayoshi** [comment]
Hi @bfallonf  , could you fix the indent of ocp 3.5 release notes?
https://github.com/openshift/openshift-docs/blob/master/release_notes/ocp_3_5_release_notes.adoc#dynamic-provisioner-for-azure-block-storage

**Comment by bfallonf** [comment]
I've updated the rhel notes directly in the github branch.

But I've also updated the docs as per this BZ and [BZ1422109](https://bugzilla.redhat.com/show_bug.cgi?id=1422109).

I'll put the BZ on QA and see if there's any feedback.

File: install_config/configuring_azure.adoc [hunk 1]
```diff
@@ -19,6 +19,30 @@ Azure disk as persistent storage] for application data. After Azure is
...
 
+[[azure-configuration-file]]
+== The Azure Configuration File
+
+Configuring {product-title} for Azure requires the *_/etc/azure/azure.conf_* file, on each node host.
+
+If the file does not exist, create it, and add the following:
+
+----
+tenantId: <> <1>
```
Comment (rootfs) [inline install_config/configuring_azure.adoc]: This file is in json format. Can you reformat it?

File: install_config/configuring_azure.adoc [hunk 2]
```diff
@@ -19,6 +19,30 @@ Azure disk as persistent storage] for application data. After Azure is
...
+aadClientSecret: <> <4>
+aadTenantId: <> <5>
+resourceGroup: <> <6>
+location: <> <7>
+----
+<1> The AAD tenant ID for the subscription that the cluster is deployed in.
+<2> The Azure subscription ID that the cluster is deployed in.
+<3> The client ID for an AAD application with RBAC access to talk to Azure RM APIs.
+<4> The client secret for an AAD application with RBAC access to talk to Azure RM APIs.
+<5> Ensure this is the same as tenant ID.
```
Comment (rootfs) [inline install_config/configuring_azure.adoc]: it is optional

File: install_config/configuring_azure.adoc [hunk 3]
```diff
@@ -19,6 +19,30 @@ Azure disk as persistent storage] for application data. After Azure is
...
+<6> The Azure Resource Group name that Azure VM belongs to.
+<7> The compact style Azure region, for example `southeastasia`.
```
Comment (rootfs) [inline install_config/configuring_azure.adoc]: it is optional

**Comment by bfallonf** [comment]
@rootfs Thanks for the review. I'm not sure why I need to reformat the file example. I think it is already in YAML format. Are you saying it should be in JSON format? I've tried to use a formatter to change it, but it's giving errors.

I've also updated the examples to say the two are optional.

If there's anything else, please let me know.

File: install_config/configuring_azure.adoc (hunk 1 above)
Comment (weherdh) [inline install_config/configuring_azure.adoc]: Json format is like below:
```
{
  "aadClientID" : ...,
  "aadClientSecret" : ...,
  "subscriptionID" : ...,
  "tenantID" : ...,
  "resourceGroup": ...,
}
```

**Comment by bfallonf** [comment]
Confirmed that YAML is fine for the file layout.

@adellape @ahardin-rh @bmcelvee @mburke5678 @gaurav-nelson peer review plz?

**Comment by bfallonf** [comment]
For reference, I added information about the azure.conf file and added a prerequisites section.

**Comment by rootfs** [comment]
lgtm

**Comment by ahardin-rh** [comment]
LGTM! :fireworks:

**Comment by bfallonf** [comment]
Thanks @ahardin-rh . I'll merge.

**Comment by bfallonf** [comment]
[rev_history]
|xref:../install_config/configuring_azure.adoc#install-config-configuring-azure[Configuring For Azure]
|Added the xref:../install_config/configuring_azure.adoc#azure-configuration-file[The Azure Configuration File] section and some file example fixes.
%
|xref:../install_config/persistent_storage/persistent_storage_azure.adoc#install-config-persistent-storage-persistent-storage-azure[Configuring Persistent Storage -> Persistent Storage Using Azure Disk]
|Added the xref:../install_config/persistent_storage/persistent_storage_azure.adoc#azure-prerequisites[Prerequisites] section.
%

**Pull request merged**